### PR TITLE
Add ginkgo unit tests for configMaps

### PR DESCRIPTION
**Pull request by Rizwana777**
**What does this PR do / why we need it**:
- Ginkgo tests for configMaps
- minor updates in other test files

**Which issue(s) this PR fixes**:
Fixes #? Issue #22  and Issue #25 and #43

### CODE REVIEW

**Review by jgwest — CHANGES_REQUESTED** [review]
In addition, I happened to notice that plugin download doesn't work on OpenShift (unfortunately we're not running the E2E tests against OpenShift, so it doesn't automatically catch this yet :sweat_smile:). It would be good to fix it is as part of this PR. To reproduce:

1) Log in to OpenShift cluster from clusterbot (e.g. `rosa create 4.14`)
2) In rollouts manager dir, run `hack/run-upstream-argo-rollouts-e2e-tests.sh`

You will see it gets stuck on this step:
```
+ kubectl get -n argo-rollouts deployment/argo-rollouts
NAME            READY   UP-TO-DATE   AVAILABLE   AGE
argo-rollouts   0/1     1            0           0s
+ kubectl wait --for=condition=Available --timeout=10m -n argo-rollouts deployment/argo-rollouts
```


If you look at the logs of the `argo-rollouts` Pod, you will see the error:

```
[jgw@localhost-lan argo-rollouts-manager]$ k logs argo-rollouts-65b44974f6-6n9dl
time="2024-02-21T06:55:05Z" level=info msg="Argo Rollouts starting" version=v1.6.6+737ca89
time="2024-02-21T06:55:05Z" level=info msg="Using namespace argo-rollouts"
time="2024-02-21T06:55:05Z" level=info msg="GET https://172.30.0.1:443/api/v1/namespaces/argo-rollouts/secrets?labelSelector=istio.argoproj.io%2Fprimary-cluster%3Dtrue&limit=1 200 OK in 27 milliseconds\n"
time="2024-02-21T06:55:05Z" level=info msg="GET https://172.30.0.1:443/version 200 OK in 1 milliseconds\n"
time="2024-02-21T06:55:05Z" level=info msg="Creating event broadcaster"
time="2024-02-21T06:55:05Z" level=info msg="Setting up event handlers"
time="2024-02-21T06:55:05Z" level=info msg="Setting up experiments event handlers"
time="2024-02-21T06:55:05Z" level=info msg="Setting up analysis event handlers"
time="2024-02-21T06:55:05Z" level=info msg="GET https://172.30.0.1:443/api/v1/namespaces/argo-rollouts/configmaps/argo-rollouts-config 200 OK in 10 milliseconds\n"
time="2024-02-21T06:55:05Z" level=fatal msg="Failed to download plugins: failed to create plugin folder for plugin (argoproj-labs/openshift-route-plugin): (mkdir /home/argo-rollouts/plugin-bin: permission denied)"
```

To fix this, in `deployment.go`, you can probably modify our generation of the Deployment, so that the Deployment contains the following volumeMounts/volume fields:

```
kind: Deployment
  name: # (...)
  namespace: # (...)
spec:
  containers:
  - name: argo-rollouts
    volumeMounts:
    - mountPath: /home/argo-rollouts/plugin-bin
      name: tmp
  volumes:
  - emptyDir: {}
    name: tmp
```

File: controllers/configmap_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			{
+				Name:     OpenShiftRolloutPluginName,
+				Location: OpenShiftRolloutPluginPath,
+			},
+		}
+		pluginString, err := yaml.Marshal(trafficRouterPlugins)
+		Expect(err).ToNot(HaveOccurred())
+
+		desiredConfigMap.Data = map[string]string{
+			"trafficRouterPlugins": string(pluginString),
```
[the remark this answers is not in the document]
Comment (Rizwana777) [reply controllers/configmap_test.go]: Done

File: controllers/deployment_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		By("fetch the Deployment")
+		fetchedDeployment := &appsv1.Deployment{}
+		Expect(fetchObject(ctx, r.Client, a.Namespace, desiredDeployment.Name, fetchedDeployment)).To(Succeed())
+
+		By("verify that the fetched Deployment matches the desired one")
+		Expect(fetchedDeployment.Name).To(Equal(desiredDeployment.Name))
+
+	})
+
+	It("should update the deployment if it already exists", func() {
```
[the remark this answers is not in the document]
Comment (Rizwana777) [reply controllers/deployment_test.go]: Yes agreed, Updated tests

**Review by jgwest — APPROVED** [review]
LGTM, thanks @Rizwana777!

**Pull request merged**